source/Group7_GP_Agent/GPAgentSensors.hpp

```diff
@@ -0,0 +1,85 @@
+
+#pragma once
+
+#include "source/core/WorldGrid.hpp"
+
+const int WALL = 2;
```

Comment:
We will have to discuss with the world groups to see if they can expose the IDs of obstacles if we want this to be robust.

Reply:
Fair enough. 

I was also wondering about that if we can have a set of non penetrable object id's

Reply:
@ssitu at this moment I can add a `todo` there are we can come back to it. I started a thread on Discord.